### PR TITLE
Byowl

### PR DESCRIPTION
**Pull request by shashank-boyapally**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Bring Your Own Custom Workload focuses on the user to customize his own workload similar to what `kube-burner` could perform, the command to invoke BYOWL is as following. I have checked using cluster-density-v2 config files. The reason to not include flags to set env vars is there were conflicting default values, if flags were allowed.
```
sboyapal cdv2-check >> kube-burner-ocp init -c config.yml
time="2024-03-04 14:25:22" level=info msg="Checking for status of rosa job/osd-cluster-ready in namespace openshift-monitoring" file="cluster_health.go:88"
time="2024-03-04 14:25:22" level=info msg="Config file config.yml available in the current directory, using it" file="helpers.go:96"
time="2024-03-04 14:25:22" level=info msg="🔥 Starting kube-burner (byowl@4b34c43c2f8ea851f17517be61d99289db4cd60c) with UUID 4027eb22-2b2f-4c08-b4b9-b682c50d78cb" file="job.go:97"
time="2024-03-04 14:25:22" level=info msg="📈 Registered measurement: podLatency" file="factory.go:79"
...
...
...
time="2024-03-04 15:26:31" level=info msg="cdv2: Initialized 50th: 0 99th: 1000 max: 1000 avg: 23" file="pod_latency.go:242"
time="2024-03-04 15:26:31" level=info msg="cdv2: Ready 50th: 3000 99th: 12000 max: 13000 avg: 5833" file="pod_latency.go:242"
time="2024-03-04 15:26:31" level=info msg="cdv2: PodScheduled 50th: 0 99th: 500 max: 1000 avg: 14" file="pod_latency.go:242"
time="2024-03-04 15:26:31" level=info msg="cdv2: ContainersReady 50th: 3000 99th: 12000 max: 13000 avg: 5833" file="pod_latency.go:242"
time="2024-03-04 15:26:31" level=info msg="Finished execution with UUID: 4027eb22-2b2f-4c08-b4b9-b682c50d78cb" file="job.go:265"
time="2024-03-04 15:26:31" level=info msg="Garbage collecting jobs" file="job.go:278"
time="2024-03-04 15:26:31" level=info msg="Deleting 1 namespaces with label: kube-burner-job=cdv2" file="namespaces.go:67"
time="2024-03-04 15:27:18" level=info msg="👋 Exiting kube-burner 4027eb22-2b2f-4c08-b4b9-b682c50d78cb" file="helpers.go:191"
```

## Related Tickets & Documents

- Related Issue #24
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
Missing a test case here :)

**Comment by shashank-boyapally** [comment]
Sorry, Will add test case!!

File: custom-workload.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+				log.Fatalf("Error creating clientSet: %s", err)
+			}
+			openshiftClientset, err := versioned.NewForConfig(restConfig)
+			if err != nil {
+				log.Fatalf("Error creating OpenShift clientset: %v", err)
+			}
+			if !ClusterHealthyOcp(clientSet, openshiftClientset) {
+				os.Exit(1)
+			}
+			wh.Metadata.Benchmark = "custom-workload"
```
Comment (rsevilla87) [inline custom-workload.go:45]: Can we get this value from a flag?

**Comment by vishnuchalla** [comment]
Since most part of this code seems like a stub to encourage users to bring their own models, can you add some instructions in our docs for their understanding?

**Review by rsevilla87 — APPROVED** [review]
lgtm

**Pull request merged**